temporalio/client.py

```diff
@@ -820,6 +811,316 @@ def get_workflow_handle_for(
+            update,
+            arg,
+            args=args,
+            start_workflow_operation=start_workflow_operation,
+            wait_for_stage=WorkflowUpdateStage.COMPLETED,
+            id=id,
+            result_type=result_type,
+            rpc_metadata=rpc_metadata,
+            rpc_timeout=rpc_timeout,
+        )
+        return await handle.result()
+
+    # Overload for no-param start update
+    @overload
+    async def start_update_with_start_workflow(
+        self,
+        update: temporalio.workflow.UpdateMethodMultiParam[[SelfType], LocalReturnType],
+        *,
+        start_workflow_operation: WithStartWorkflowOperation[SelfType, ReturnType],
+        wait_for_stage: WorkflowUpdateStage,
+        id: Optional[str] = None,
+        rpc_metadata: Mapping[str, str] = {},
+        rpc_timeout: Optional[timedelta] = None,
+    ) -> WorkflowUpdateHandle[LocalReturnType]: ...
+
+    # Overload for single-param start update
+    @overload
+    async def start_update_with_start_workflow(
+        self,
+        update: temporalio.workflow.UpdateMethodMultiParam[
+            [SelfType, ParamType], LocalReturnType
+        ],
+        arg: ParamType,
+        *,
+        start_workflow_operation: WithStartWorkflowOperation[SelfType, ReturnType],
+        wait_for_stage: WorkflowUpdateStage,
+        id: Optional[str] = None,
+        rpc_metadata: Mapping[str, str] = {},
+        rpc_timeout: Optional[timedelta] = None,
+    ) -> WorkflowUpdateHandle[LocalReturnType]: ...
+
+    # Overload for multi-param start update
+    @overload
+    async def start_update_with_start_workflow(
+        self,
+        update: temporalio.workflow.UpdateMethodMultiParam[
+            MultiParamSpec, LocalReturnType
+        ],
+        *,
+        args: MultiParamSpec.args,  # pyright: ignore
+        start_workflow_operation: WithStartWorkflowOperation[SelfType, ReturnType],
+        wait_for_stage: WorkflowUpdateStage,
+        id: Optional[str] = None,
+        rpc_metadata: Mapping[str, str] = {},
+        rpc_timeout: Optional[timedelta] = None,
+    ) -> WorkflowUpdateHandle[LocalReturnType]: ...
+
+    # Overload for string-name start update
+    @overload
+    async def start_update_with_start_workflow(
+        self,
+        update: str,
+        arg: Any = temporalio.common._arg_unset,
+        *,
+        start_workflow_operation: WithStartWorkflowOperation[SelfType, ReturnType],
+        wait_for_stage: WorkflowUpdateStage,
+        args: Sequence[Any] = [],
+        id: Optional[str] = None,
+        result_type: Optional[Type] = None,
+        rpc_metadata: Mapping[str, str] = {},
+        rpc_timeout: Optional[timedelta] = None,
+    ) -> WorkflowUpdateHandle[Any]: ...
+
+    # TODO (dan):
+    # temporalio/client.py:926: error: Overloaded function implementation does not accept all possible arguments of signature 1  [misc]
+    async def start_update_with_start_workflow(  # type: ignore
```

Comment:
I think here and other user facing entry points (e.g. execute equivalent and the class doc for `WithStartWorkflowOperation`) should have the "experimental" warning that looks similar to what we're removing on #707.

Reply:
Added experimental warnings